Change Member Location

s9latimm opened the pull request:
Related https://github.com/se-sic/vara-llvm-project/pull/55
Related https://github.com/se-sic/VaRA/pull/906

Closes https://github.com/se-sic/VaRA/issues/879

s9latimm committed:
Refactor FeatureMemberOffset

M	include/vara/Feature/FeatureSourceRange.h
M	lib/Feature/FeatureModelParser.cpp
M	unittests/Feature/FeatureModelParser.cpp
M	unittests/Feature/FeatureSourceRange.cpp

s9latimm committed:
Reformat

M	include/vara/Feature/FeatureSourceRange.h

s9latimm committed:
Improve data structure

M	include/vara/Feature/FeatureSourceRange.h
M	unittests/Feature/FeatureModelParser.cpp
M	unittests/Feature/FeatureSourceRange.cpp

s9latimm committed:
Reverse nesting

M	include/vara/Feature/FeatureSourceRange.h
M	unittests/Feature/FeatureSourceRange.cpp

vulder commented on include/vara/Feature/FeatureSourceRange.h:
```diff
@@ -72,70 +72,72 @@ class FeatureSourceRange {
+    FeatureMemberOffset(const llvm::SmallVector<llvm::StringRef, 1> &Class,
+                        llvm::StringRef Member)
+        : Member(Member.str()) {
+      for (const auto C : Class) {
+        this->Class.push_back(C.str());
+      }
```
```suggestion
```

vulder commented on include/vara/Feature/FeatureSourceRange.h:
```diff
@@ -72,70 +72,72 @@ class FeatureSourceRange {
-    std::string MemberOffset;
+    FeatureMemberOffset(const llvm::SmallVector<llvm::StringRef, 1> &Class,
+                        llvm::StringRef Member)
+        : Member(Member.str()) {
```
```suggestion
        : Class{Class.begin(), Class.end()} ,Member(Member.str()) {
```
This correctly transfers the items of the SmallVector

vulder commented on include/vara/Feature/FeatureSourceRange.h:
```diff
@@ -72,70 +72,72 @@ class FeatureSourceRange {
-      if (isInMemberOffsetFormat(MemberOffset)) {
-        return FeatureMemberOffset(std::move(MemberOffset));
+    static std::optional<FeatureMemberOffset>
+    createFeatureMemberOffset(llvm::StringRef MemberOffset) {
```
Please put the "splitting" semantics in a helper function, so we have a better abstraction.

vulder reviewed (changes requested):
Except for the things we discussed LGTM

s9latimm committed:
Move splitting into abstraction

M	include/vara/Feature/FeatureSourceRange.h
M	unittests/Feature/FeatureSourceRange.cpp
